### ISSUE TITLE
nbqa-pylint cell number does not include markdown cells

### ISSUE THREAD

**Issue by rabyj** [nbQA-dev/nbQA#860]
**Issue**

When getting warnings from nbqa-pylint, e.g. `notebook.ipynb:cell_XX:Y:Z: C0206: Consider iterating with .items() (consider-using-dict-items)`, the cell number does not refer to the same number as in vscode. cell_XX does not count markdown cells, rendering the message confusing (and way less useful in large notebooks with lots of markdown)

Joining an example notebook as exported script, since github won't allow me to joint it directly. 

I run `pre-commit run nbqa-pylint` on the staged notebook to generate warnings.

**System information**
```text
OS: Ubuntu 20.04.6 LTS
vscode: 1.92.1
ms-toolsai.jupyter: 2024.7.0
nbqa: 1.8.7
pylint: 2.16.4
pre-commit: 3.0.4
```

**Notebook exported as Python script**
```Python
# %%
"""Demonstrating a nbqa-pylint issue."""

# %%
a_dict = {"test":1}

# %%
# forced C0206: vscode cell: 3, nbqa-pylint cell: 3
for key in a_dict:
    b = a_dict[key]

# %% [markdown]
# markdown cell: miaw

# %% [markdown]
# markdown cell 2: miaw two, electric boogaloo

# %%
# forced C0206: vscode cell: 6, nbqa-pylint cell: 4
for key in a_dict:
    b = a_dict[key]

# %%
# forced C0206: vscode cell: 7, nbqa-pylint cell: 5
for key in a_dict:
    b = a_dict[key]
```

Side question I couldn't find an answer to: Is there a way to force nbqa-pylint to show the context of the error and/or cell content, instead of just showing the warning with cell number? I can file it somewhere else (where?) if this is inappropriate.

**Comment by MarcoGorelli** [comment]
Hey @rabyj 👋 

thanks for taking the time to report this!

in Jupyter Lab, the numbers line up for me:

![image](https://github.com/user-attachments/assets/b7a34ee7-37c1-4093-85f4-bedbd7d53ba5)

If vscode is showing numbers differently, then maybe we need ask them to reconsider their numbering?

We _could_ introduce an option in nbQA to let the user choose, although:
- I predict that ruff will subsume nbQA completely before too long (r.i.p. my lockdown project)
- I don't have much time for this project at the moment so can't really prioritise adding anything new. Happy to review a pull request if you have interest / time though

**Comment by rabyj** [comment]
Thanks for the fast reply.

Isn't that the execution number though, which will change a lot when using a notebook? afaik Jupyter Lab does not actually number cells.

vscode also has the same execution number, but I haven't been able to use that reliably to reference nbqa-pylint cell number because I'd have to re-run everything from the start each time I want the cell number

**Comment by MarcoGorelli** [comment]
true, I think I was assuming that people would be checking notebooks into version control after having done "restart and run all" (say, if they're using notebooks for documentation)
